### ISSUE TITLE
{Initial Feature: User Sign-Up} Task Documentation Creation

### ISSUE THREAD

**Issue by pierreg-vales** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#58]
Description:

Create a detailed documentation for your task to preemptively address potential issues. This will help prevent confusion, delays, or errors in the future by providing a clear roadmap of the task's objectives and completion criteria.

Requisites: 
- A detailed breakdown of the task: Explain each step involved in the task, why it's being done, and its importance.
- Prerequisites: Any resources, permissions, or dependencies required to complete the task.
- Expected outcomes: What the final result should look like and any milestones or deliverables along the way.
- Potential risks or challenges: Identify any obstacles that might occur during the task and how they could be mitigated.
- Roles and responsibilities: Clarify who is responsible for each aspect of the task if it's a team effort.

Criteria for Completion:
- Document must be well-structured and easy to follow
- It should provide sufficient information to create a new issue or task from scratch with minimal guidance.
- The documentation should contain clear and specific instructions on how to proceed, including links to relevant resources or tools if necessary.
- The task is considered complete when the documentation has been reviewed and approved by the  team leader/manager

**Comment by estefania-2817** [comment]
Description:

- Build the user interface of the sign up page for students and organizations.
- In case of student, build the basic structure of the sign up form where the user provides their student id, email, password and password confirmation. 
-  In case of organization, build the basic structure of the sign up form where the organization provides the name of organization, email, password and password confirmation. 
- Create the buttons where the each of them can submit their solicited information. 

* Include the link in the case if the user or organization already has been registered and only needs to log in.  

Requirements: 
For students, create the fields of:
* student id
* email
* password
* password confirmation
* log in link
* submit(Create account) button.
For organizations, create the fields of:
* organization name
* email
* password
* password confirmation
* log in link
* submit(Create account) button.

**Comment by AlejandroClaudio** [comment]
Description:
Researching on Implementing a database using Flask and SQLite3, with or without a dummy database. The goal is to research and set up a functioning system that can handle user data, either through a dummy database for testing or a real SQLite3 database for production. This system will also include JWT (JSON Web Tokens) to manage user sessions and authentication across multiple requests.

Requisites:
•	The purpose of the task is researching a way to have a functional Flask backend that communicates with an SQLite3 database for storing and retrieving user data. Flask provides the web framework, while SQLite3 is a lightweight database.
•	install Flask and SQLite3 in your Python environment.
•	User Authentication with JWT since it allows for secure authentication, ensuring that users don't have to log in repeatedly during a session. This is especially important if OAuth is used or if user sessions need to persist.
•	Have SQLite3 database (or create a dummy database) for storing user data, such as, emails passwords, and other relevant information.
•	Set up a database interface to interact with the SQLite3 database. Create a dbinterface.py file that will handle the database interactions
•	Implement routes for handling user sign-up since users need to interact with the system 
•	Use dummy data in dbinterface.py test real database interaction until the actual database is connected.

Prerequisites:
•	Flask must be installed and set up.
•	SQLite3 should be available, either as a dummy or real database.
•	Have Python modules for JWT installed.
•	Permissions and access to necessary resources, such as databases, must be available.

Expected Outcomes:
•	A functional Flask app that interacts with either a dummy or real SQLite3 database.
•	The app should be able to handle user sign-up and data retrieval.
•	With JTW users will be able to sign in and authenticate themselves without repeatedly entering credentials.

Challenges
•	User data (especially passwords) could be stored in an insecure way. Hashing password will help mitigate.
•	JWT tokens could be compromised or leaked, so a token expiration and reset should help the issue.
•	Database connections could be an issue, so using exception handling and log errors properly will help identify and resolve the problem.

Criteria for Completion:
•	The app should be able to store and retrieve user data from SQLite3, either as a dummy database or the real one.
•	JWT-based authentication should allow for more secure user authentication and session management.

Discussions about database resources by Ojani: #67 and #68 

This task could be shortened and/or split into more tasks.
